src/controllers/users.controller.js

```diff
@@ -263,6 +267,127 @@ class UserController {
     const allUsers = await db.user.findAll(options);
     return Responses.handleSuccess(200, 'successfully retrieved all users', res, allUsers);
   }
+
+  /**
+   * Upload travel documents
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} response
+   */
+  async uploadDocument(req, res) {
+    const {
+      name,
+    } = req.body;
+
+    let url = '';
+    if (req.file !== undefined) {
+      const {
+        filename,
+        path,
+        mimetype
+      } = req.file;
+
+      url = await filesService.s3Upload({ path, filename, mimetype }, 'documents');
+    }
+
+    const { userId } = res.locals.user;
+
+    const document = await UserService.addDocument({ name, url, userId });
+    Responses.handleSuccess(201, 'Document uploaded successfully', res, { name, url, userId });
+  }
+
+  /**
+   * Delete travel document
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} response
+   */
+  async deleteDocument(req, res) {
+    const { id } = req.params;
+    const { userId } = res.locals.user;
+
+    const document = await UserService.getDocument(id);
+
+    const isOwner = document.userId === userId;
+
+    if (isOwner) {
+      await UserService.deleteDocument(id);
+      return Responses.handleSuccess(200, 'Document Deleted successfully', res);
+    }
+    return Responses.handleError(409, 'Document doesn\'t belong to you', res);
+  }
+
+
+  /**
+   * Retrieve travel document
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} response
+   */
+  async getDocuments(req, res) {
+    const { userId, role } = res.locals.user;
+
+    const documents = await UserService.retrieveDocuments({
+      ...role === 'requester' && { userId },
+      ...role === 'travel_administrator' && { travelAdminId: userId },
+      ...role === 'suppliers' && { travelAdminId: userId },
```

Comment:
I prefer simplicity to elegance. Please make this a bit more straightforward.

Reply:
thanks @kipropbrian, let me refactor this!

Reply:
Here is how I refactored the those lines, what do you think @kipropbrian ??
```
const requesterId = role === 'requester' && userId;
const` documents = await UserService.retrieveDocuments({ requesterId });
```